### PR TITLE
[Android] Public API XWalkUIClient.onJsAlert, XWalkUIClient.onJsConfirm, XWalkUIClient.onJsPrompt and implement their test cases

### PR DESCRIPTION
**Pull request by shaoboyan**
Public API XWalkUIClient.onJsAlert, XWalkUICLient.onJsConfirm and
XWalkUIClient.onJsPropmt for client to customize these dialogs.
Implement test cases for XWalkUIClient.onJsAlert, XWalkUIClient.onJsConfirm,
and XWalkUIClient.onJsPrompt.

BUG=XWALK-4917

### CODE REVIEW

**Comment by shaoboyan** [comment]
@lincsoon @fujunwei  Plesae help to review this patch, thx.

**Comment by fujunwei** [comment]
LGTM.

**Comment by shaoboyan** [comment]
ping @lincsoon PTAL

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkUIClientInternal.java [hunk 1]
```diff
@@ -381,8 +381,21 @@ public void onPageLoadStarted(XWalkViewInternal view, String url) {
     @XWalkAPI
     public void onPageLoadStopped(XWalkViewInternal view, String url, LoadStatusInternal status) {
     }
-
-    private boolean onJsAlert(XWalkViewInternal view, String url, String message,
+    
+    /**
+     * Tell the client to display an alert dialog to the user.
+     * WARN: Please avoid to override this API and onJavascriptModalDialog API in the 
+     *       same subclass to avoid unexpected behavior. If cannot avoid, Please
```
Comment (lincsoon) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkUIClientInternal.java]: Just don't let the user do that, don't be ambiguous. Use assert expression to forbid it

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkUIClientInternal.java [hunk 2]
```diff
@@ -381,8 +381,21 @@ public void onPageLoadStarted(XWalkViewInternal view, String url) {
...
+    /**
+     * Tell the client to display an alert dialog to the user.
+     * WARN: Please avoid to override this API and onJavascriptModalDialog API in the 
+     *       same subclass to avoid unexpected behavior. If cannot avoid, Please 
+     *       remember to call onJsAlert in the overriden onJavascriptModalDialog
+     *       method for proper behavior.
+     * @param view the owner XWalkViewInternal instance.
+     * @param url the url of the web page which wants to show this dialog.
+     * @param message the message to be shown.
+     * @param result the callback to handle the result from caller.
```
Comment (lincsoon) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkUIClientInternal.java]: No comment about return value

**Comment by lincsoon** [comment]
You can merge 3 test files of JS to only one file and make 3 test cases

**Comment by shaoboyan** [comment]
Sorry for some mass , I'll pull request another PR to replace this one.

**Comment by shaoboyan** [comment]
Sorry for some mess, I'll close this pr and push another one to replace it

**Comment by shaoboyan** [comment]
Sorry for some mass, I'll close this pr and re pull another one for replacing.

**Pull request closed without merging**